### PR TITLE
encode refundETH on swaps with very high slippage with ETH input token.

### PR DESCRIPTION
**Pull request by ewilz**
This will help ensure that as pools reach min/max price, causing a partialFill on exactInput, the leftover ETH gets refunded. There may still exist even edgier cases where it _may_ get left off, but we believe this should account for 99% of this scenario.

### CODE REVIEW

File: src/swapRouter.ts [hunk 1]
```diff
@@ -469,6 +467,22 @@ export abstract class SwapRouter {
     }
   }
 
+  // if slippage is very high, there's a chance of hitting max/min prices resulting in a partial fill of the swap
+  private static riskOfPartialFill(trades: AnyTradeType): boolean {
+    const FIFTY_PERCENT = new Percent(JSBI.BigInt(50), JSBI.BigInt(100))
```
Comment (willpote) [inline src/swapRouter.ts:472]: Nit: constant at top of file, and named more descriptively like: `REFUND_ETH_PRICE_IMPACT_THRESHOLD`

File: src/swapRouter.ts [hunk 2]
```diff
@@ -469,6 +467,22 @@ export abstract class SwapRouter {
...
+      return trades.priceImpact.greaterThan(FIFTY_PERCENT)
+    } else if (Array.isArray(trades)) {
+      for (const trade of trades) {
+        if (trade instanceof V3Trade && trade.priceImpact.greaterThan(FIFTY_PERCENT)) {
+          return true
+        }
+      }
+    }
+    return false
+  }
```
Comment (willpote) [inline src/swapRouter.ts:484]: We need to check tokenIn is native eth too

Could you also add a test that checks the calldata is only added for native eth too

Comment (willpote) [reply src/swapRouter.ts:484]: Also nit:

```
if (!Array.isArray(trades)) {
    trades = [ trades ]
}

return _.some(trades, trade => trade.priceImpact.greaterThan(..))
```

Comment (ewilz) [reply src/swapRouter.ts:484]: > We need to check tokenIn is native eth too

doesn't this happen on line 363?

Comment (willpote) [reply src/swapRouter.ts:484]: Whoops, yes

**Review by willpote — APPROVED** [review]

**Pull request merged**